bgpd/bgp_aspath.c

```diff
@@ -414,6 +414,19 @@ unsigned int aspath_count_hops(const struct aspath *aspath)
 	return count;
 }
 
+/* Check if aspath has AS_SET or AS_CONFED_SET */
+int aspath_check_as_sets(struct aspath *aspath)
```

Comment:
Would the pattern be better if we passed in the `struct bgp *bgp` and just had this function do the right thing based upon the configuration?  Instead of having to have 2 if statements in a bunch of other places

Reply:
I'd also prefer that we returned bool instead of int

Reply:
Will fix this

Reply:
>Would the pattern be better if we passed in the struct bgp *bgp and just had this function do the right thing based upon the configuration? Instead of having to have 2 if statements in a bunch of other places

Would you mind giving an example of what did you have in your mind? This function is called only from 2 places: subgroup_announce_check() and bgp_update() like other checks.